gems/smithy-client/lib/smithy-client/handlers.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    # A collection of generic handlers that are used by Smithy Clients.
+    # @api private
+    module Handlers
```

Comment:
I'm not a fan of the half and half. Mainly because plugin code is not tied to the handler in the same package, so drift can occur if we reorder or change code, making extra cases to consider. I still think the entirety of the protocol plugin can be generic. The alternative is fully code generated which you had before. The only part that needs to be code generated is configuration default right? I am also thinking about the endpoint case on how we can genericize that one. One option I don't think we considered would be to set default config values using the before_initialize hook, perhaps in a code generated default config plugin. We can chat offline on it.

Reply:
Agreed, the separation between codegeneration plugin that using these handlers and the handlers definition in a separate file does not feel "right" to me. Your idea on having a codegenerated default config plugin is a good idea. 

In the subject of codegenerated endpoint plugin, I was think making the Endpoint Handler class generic could be tricky since it uses the codegenerated EndpointsParameters defined within the gem: https://github.com/smithy-lang/smithy-ruby/blob/decaf/projections/weather/lib/weather/plugins/endpoint.rb#L29

(Could be different if we are able to make that portion generic - however)

Reply:
The issue with differing versions of codegen vs client runtime is a good point on this.

I'd suggest the entire plugin be code generated in that case.

Runtime resolution of the default plugin doesn't seem like the right behavior to me.

> One option I don't think we considered would be to set default config values using the before_initialize hook, perhaps in a code generated default config plugin. We can chat offline on it.

This is interesting - but how is it different from setting a default block in an `option` (except for the fact we couldn't generate default doc option, which I think is important for customers).

Reply:
Let's have a discussion offline either later today or tomorrow AM.